File: test-utils/src/main/java/ru/virgil/testutils/fluent_request/BodyStepJsonStart.java
package ru.virgil.testutils.fluent_request;

public interface BodyStepJsonStart {

    BodyStepJson send(Object dto);

    BodyStepJson receive(Class<?> responseClass, Class<?>... responseClasses);

    BodyStepJson receiveAsBytes();

    BodyStepJson exchange(Object dto, Class<?> responseClass, Class<?>... responseClasses);

    FinalStep<?> and() throws Exception;
}

File: tools/src/main/kotlin/ru/virgil/spring_tools/tools/testing/fluent_request/BodyStepJsonStart.kt
package ru.virgil.spring_tools.tools.testing.fluent_request

interface BodyStepJsonStart {

    fun send(dto: Any): BodyStepJson
    fun receive(responseClass: Class<*>, vararg responseClasses: Class<*>): BodyStepJson
    fun receiveAsBytes(): BodyStepJson
    fun exchange(dto: Any, responseClass: Class<*>, vararg responseClasses: Class<*>): BodyStepJson

    @Throws(Exception::class)
    fun and(): FinalStep<*>
}
